### DISCUSSION TITLE
Add text ad end of date

### DISCUSSION THREAD

**Discussion by filmgal** [liamcain/obsidian-calendar-plugin#90] [Ideas]
sorry - I just noticed a similar request already in Issues:  Wildcard in note title. Please ignore this.

Hi there. Thanks so much for this plug-in. I love it.

Would it be possible to recognize Daily Notes that include text after the date? My files are named 20201225 as per my template. I like to add some text to the filename after the date, either to capture something about the day or to act as a reminder. For example:

20201225 fixed the template bug
20201226 planted hydroponic lettuce
20201227 call Pete for his birthday

The minute I add text to the filename the file is no longer recognized by Calendar. Perhaps there is already an existing something or other that I could add at the end of the date in my template that would work as a stand-in for additional text? If not, then please consider this a feature request if you think it would be useful for others as well.

**Comment by liamcain** [comment]
Hi @filmgal, thanks for the kind words!

Since this isn't a straightforward change, I'd love to understand your use case a little bit more. I also have some workflow suggestions in case that helps.

## Workflow Questions
1.) Is the text just to help spot the day in the file explorer? Or is it for linking? (i.e. finding the day when typing `[[pete's birthday...`
2.) Do you only have 1 note per day? Or could their be multiple?

Q2 is my biggest concern if the calendar supports wildcards. It could lead to ambiguity around what note to open if there are multiple matching the same day.

## Suggestions

> **Note:** These are just some aspects of my workflow that I think could fill the gap you're looking for. But I'm still interested in hearing more about your workflow to see if there's a match!

1.) Daily note as an **anchor**. This is the workflow that I do: I always have a daily note in the format of `YYYY-MM-DD`, then for spin-off notes, I prefix them with the date as well.  So for your example, you would have `20201225.md` and that file would link to `20201225 fixed the template bug`.

2.) Note **aliases** for days. I mentioned this in another related feature request [here](https://github.com/liamcain/obsidian-calendar-plugin/issues/39#issuecomment-732209551). In your daily note (`YYYY-MM-DD`), you could add `20201225 fixed the template bug` as an alias to the note. This might not totally fit your workflow since (making an assumption here) aliases don't appear in the File Explorer, but it does help for things like `alias: Pete's birthday`.

Let me know if that helps at all!

**Reply by filmgal** [reply]
Hi Iiamcain,

Thanks for the reply.  Ah, you bring up an issue I hadn't really considered, but yes, I do have multiple notes beginning with the same date and I can see how that would cause an issue.  I suppose what I really need is a different kind of calendar: one that would display all notes that include 20201225 in the title (or better yet, in the title and in the text).

I have tried some alternative workarounds. I've created the single note and added links to my other notes, but that turns out to be too much work for me. And I must confess I don't quite grasp the alias business. I think I would need an "alias for dummies" tutorial to figure that one out.

To answer your question #1: I like to add more than the date so that when I look at explorer outside Obsidian (the way I have to do on my Android devices because stuff is in Dropbox) I need to see 20201225 Gardening vs 20201225 fixed the bug vs 20201225 did something or other.  I love to keep track of the dates I've done things, but seeing a folder with a bunch of dates doesn't give me enough information.

So the answer to question 1 is: to spot the day in explorer
and the answer to question 2 is: multiples, precisely because of question 1.

**Reply by liamcain** [reply]
Aah, okay. I understand. It's not so much that you're interested in adding text to the end of daily notes. It's that you're interested in supporting Zettelkasten-prefixed notes (and you may or may not have a daily note). I think that's totally reasonable. It will be a bit tricky since I don't have an interface in mind quite yet, but I'll like this discussion open and add ideas to it as I give it more thought.

**Reply by filmgal** [reply]
I guess that's true, although I had never heard of Zettelkasten before last month, and have been using dates as prefix since the early 90s!  I do use your program and I do use those daily notes. I wonder... I'm no programmer so I have no clue if what I'm about to suggest is an entire different ballgame...currently clicking on the date in the calendar opens the daily note.  What if right-clicking on the date displayed the search results of that date? Maybe the results could be displayed in a scrollable fly-out or something, where we could select from the note we want from the flyout list?  It does sound complicated though.  Just thinking out loud...

**Reply by liamcain** [reply]
I only mention Zettelkasten (ZK) notes since that's what Obsidian refers to notes prefixed with timestamps if you use the ZK core plugin:

<img width="853" alt="image" src="https://user-images.githubusercontent.com/693981/103448124-f6706880-4c62-11eb-952c-cc42c18cdca9.png">

So keeping that in might it feels like a good idea to offer better compatibility with core plugins.

---

I like the "Show date in search results" idea! That might be a good starting point before building out a more complicated UI. Thanks for that!

**Reply by jorgesanchezurien** [reply]
Figured out a partial solution with a simple modification in the plugin source code:

- edit main.js under .obsidian/plugins/calendar
- locate the following line in funtion getDateFromFile(file, granularity)

> const noteDate = window.moment(file.basename, format, true);

- change the last argument "true" to "false" to disable strict parsing: https://momentjs.com/guides/#/parsing/strict-mode/

This shows a dot in calendar days with at least one note which filename contains a matching date and opens the first one when clicked. I guess the dots would make more sense indicating how many notes match for a given day, i.e, 'o' for one note, 'o+' for more than one...
